### ISSUE TITLE
Possible EventEmitter memory leak on guildMembersChunk when using fetchAllMembers

### ISSUE THREAD

**Issue by Yomanz** [discordjs/discord.js#3943]
**Please describe the problem you are having in as much detail as possible:**

When the bot is starting (before `shardReady` is called) 
`(node:27004) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 guildMembersChunk listeners added to [Bot]. Use emitter.setMaxListeners() to increase limit`

It only happens when `fetchAllMembers` is `true`.

This is logged to the console.
**Further details:**
I have members intent enabled and whitelisted.
This specific bot is only in 264 guilds.

- discord.js version: 12.0.1
- Node.js version: 12.13.0
- Operating system: Windows 10
- Priority this issue should have – please be realistic and elaborate if possible: It should be a top priority unless i'm being an idiot.

- [ ] I have also tested the issue on latest master, commit hash:

**Comment by LxveFades** [comment]
You're most likely emitting that event 11 times, or more, somewhere, which is causing that warning to log. 
I do not think that this is a bug, this is you spawning that event 11, or more, times somewhere in your code. 
However I could be missing the point of this completely, but I think the description was vague, at least in my opinion; which was insufficient for me to understand completely.

tl;dr: You're causing this, not the library.

**Comment by Yomanz** [comment]
I don't use that event anywhere

**Comment by LxveFades** [comment]
The library doesn't spawn events for you. The only one, out of you and the library, who could possibly spawn those events - is you.

**Comment by Yomanz** [comment]
https://github.com/discordjs/discord.js/search?q=guildMembersChunk&unscoped_q=guildMembersChunk

**Comment by LxveFades** [comment]
The library _emits_ the event, it doesn't spawn it. It emits any other event that's documented, you're the one to fire it up, however.

**Comment by Yomanz** [comment]
![image](https://user-images.githubusercontent.com/5119107/76702666-6eb17180-66c3-11ea-9371-f69918ab81ad.png)
satisifed??!?!?!?!?

**Comment by PyroTechniac** [comment]
The library attaches event listeners on GuildMemberManager#fetch(), which then gets emitted when the WebSockets receives the events from the gateway

**Comment by Yomanz** [comment]
I've updated the issue to outline the fact it only happens when `fetchAllMembers` is enabled.

**Comment by PyroTechniac** [comment]
Yes, because fetchAllMembers calls GuildMemberManager#fetch(), and it does so when it receives the guild, so when it receives all guilds, it calls that method and attached many listeners

**Comment by anandre** [comment]
It's a known thing with fetching members like that.  In the future, please don't ignore the templates.

**If you need help with discord.js installation or usage, please go to the discord.js Discord server instead:
  https://discord.gg/bRCvFy9
This issue tracker is only for bug reports and enhancement suggestions.
You won't receive any basic help here.**

**Comment by Yomanz** [comment]
> 
> 
> It's a known thing with fetching members like that. In the future, please don't ignore the templates.
> 
> **If you need help with discord.js installation or usage, please go to the discord.js Discord server instead: https://discord.gg/bRCvFy9 This issue tracker is only for bug reports and enhancement suggestions. You won't receive any basic help here.**

This is a bug report??!?!?

**Comment by SpaceEEC** [comment]
It would have been nice if you directly included your code in the issue as the template explicitly requests you to, @Yomanz.

To confirm @PyroTechniac's theory, please run node with `--trace-warnings` or log the `warning` event of `process`, the stacktrace should point to the manager in question then.
<https://nodejs.org/dist/latest-v12.x/docs/api/cli.html#cli_trace_warnings>
<https://nodejs.org/api/process.html#process_event_warning>

**Comment by Yomanz** [comment]
![image](https://user-images.githubusercontent.com/5119107/76707315-032ccb80-66e6-11ea-95d9-759eddc76869.png)

I didn't have a copyable reproducible bit of code at the start, I only made that during.
